internal/graph/reachableresources.go

```diff
@@ -376,6 +376,13 @@ func (crr *CursoredReachableResources) lookupTTUEntrypoint(ctx context.Context,
 		OptionalSubjectIds: req.SubjectIds,
 	}
 
+	// Optimization: if there is a single allowed relation, pass it as a subject relation filter to make things faster
+	// on querying.
+	if allowedRelations.Len() == 1 {
+		allowedRelationName := allowedRelations.AsSlice()[0]
+		subjectsFilter.RelationFilter = datastore.SubjectRelationFilter{}.WithRelation(allowedRelationName)
+	}
```

Comment:
We could, yes, but I suspect that there is a diminished state of return there, because then we still need the field in the cursor. The primary goal here was to remove the field from the cursor, which occurs if and only if we lock it to a single value